File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.cpp
class Solution {
public:
    bool isVowel(char &c){
        return (c=='A' || c=='E' || c=='I' || c=='O' || c=='U' || c=='a' || c=='e' || c=='i' || c=='o' || c=='u');
    }
    string sortVowels(string s) {
        vector<char> v;
        for(int i=0; i<s.size(); i++){
            if(isVowel(s[i])) v.push_back(s[i]);
        }
        sort(v.begin(), v.end());
        for(int i=0, j=0; i<s.size(); i++){
            if(isVowel(s[i])) s[i] = v[j++];
        }
        return s;
    }
};

File: 2785-sort-vowels-in-a-string/2785-sort-vowels-in-a-string.java
class Solution {
    boolean isVowel(Character c){
        return (c=='A' || c=='E' || c=='I' || c=='O' || c=='U' || c=='a' || c=='e' || c=='i' || c=='o' || c=='u');
    }
    public String sortVowels(String s) {
        ArrayList<Character> temp = new ArrayList<>();
        for(char c : s.toCharArray()){
            if(isVowel(c)) temp.add(c);
        }
        Collections.sort(temp);
        StringBuilder ans = new StringBuilder();
        for(int i=0, j=0; i<s.length(); i++){
            if(isVowel(s.charAt(i))) ans.append(temp.get(j++));
            else ans.append(s.charAt(i));
        }
        return ans.toString();
    }
}